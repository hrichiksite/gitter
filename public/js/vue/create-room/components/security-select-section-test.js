const mount = require('../../__test__/vuex-mount');
const { default: SecuritySelectSection } = require('./security-select-section.vue');
const { types } = require('../store');

describe('SecuritySelectSection', () => {
  it('matches snapshot', () => {
    const { wrapper } = mount(SecuritySelectSection);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('clicking public radio fires store mutation', () => {
    const { wrapper, stubbedMutations } = mount(SecuritySelectSection, {}, store => {
      store.state.createRoom.roomSecurity = 'PRIVATE';
    });

    wrapper.find({ ref: 'publicSecurityRadio' }).trigger('click');

    expect(stubbedMutations.createRoom[types.SET_ROOM_SECURITY]).toHaveBeenCalledWith(
      expect.anything(),
      'PUBLIC'
    );
  });

  it('clicking private radio fires store mutation', () => {
    const { wrapper, stubbedMutations } = mount(SecuritySelectSection, {}, store => {
      store.state.createRoom.roomSecurity = 'PUBLIC';
    });

    wrapper.find({ ref: 'privateSecurityRadio' }).trigger('click');

    expect(stubbedMutations.createRoom[types.SET_ROOM_SECURITY]).toHaveBeenCalledWith(
      expect.anything(),
      'PRIVATE'
    );
  });
});
