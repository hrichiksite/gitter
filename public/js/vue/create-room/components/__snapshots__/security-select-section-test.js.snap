// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecuritySelectSection matches snapshot 1`] = `
<section
  class="security-select-section"
>
  <ul
    class="security-list"
  >
    <li
      class="security-list-item"
    >
      <input
        class="security-list-item-radio"
        id="public-security"
        name="security"
        type="radio"
        value="PUBLIC"
      />
       
      <label
        class="security-list-item-label"
        for="public-security"
      >
        <i
          class="security-list-item-icon octicon octicon-globe"
        />
         
        <span>
          <span
            class="security-list-item-name"
          >
            Public
          </span>
           
          <br />
           
          <span>
            
            Anyone in the world can join.
          
          </span>
        </span>
      </label>
    </li>
     
    <li
      class="security-list-item"
    >
      <input
        class="security-list-item-radio"
        id="private-security"
        name="security"
        type="radio"
        value="PRIVATE"
      />
       
      <label
        class="security-list-item-label"
        for="private-security"
      >
        <i
          class="security-list-item-icon octicon octicon-lock"
        />
         
        <span>
          <span
            class="security-list-item-name"
          >
            Private
          </span>
           
          <br />
           
          <span>
            
            Only people added to the room can join.
          
          </span>
        </span>
      </label>
    </li>
  </ul>
</section>
`;
